Extensions for Scratch 2

ruairiohagan opened the pull request:
Additional python "Helper app" client to allow block extensions to be added to Scratch 2

Added ScratchServer.py to the client folder, along with a"Scratch"  folder containing the files needed to add an extension to Scratch 2.

For instructions on how to add the extension to Scratch 2 app, see Scratch\ScratchReadme.txt

SuhaylZhao commented:
Thank you very much for your excellent work! 

We've tested your code. Except for two bugs and one question, it's perfect.

Bugs(In file "ScratchServer.py"):
1. In line 97, the code should be "return self. buzz (pathParts [2])"
2. In line 99, the code should be "return self. buzzWait (pathParts [2])"

question:
It seems that there is no camera in your program, but it occupies the port of the camera, which leads to the conflict between the program and the camera service. Why?

About Pull requests

This repertory is our product "Raspberry Pi Car" repository, we are responsible for all the contents of this repertory. so all submissions should be carried out by us (Freenove). Maybe we should build a new repertory that allows all contributors to submit. 

For your contribution, we decided to put it in file "tutorial and readme" to illustrate it and provide a link to your repertory. Do you agree?

Thanks again！

ruairiohagan commented:
Hi Suhayl,

Thanks for your great feedback!

I've updated the files to remove the buzz bugs you spotted and also changed the port the helper app runs under to 8085 instead of 8090 (my bad, I just missed that the camera was using that port!).

I fully accept that you want to keep control of the main branch, putting the link in the tutorial / readme is good idea,

I just wanted to share with others how to plug in your great little robot kit into scratch, I'm really happy you are going to link it into the official guide.

If you set up the shared repository you mentioned please let me know and I'll submit the pull request to that.

Many thanks,

Ruairi.
